### ISSUE TITLE
Tile classification -> annotations not working

### ISSUE THREAD

**Issue by saramcardle** [qupath/qupath#359]
Sorry to keep bugging you with more bugs :/

In m4, the tile classification -> annotations feature stopped working. I've tested it on brightfield images only, with rectangular tiles and both types of superpixels. When you use the plugin, no new annotations are created, however internal annotations and the tiles themselves are deleted (when those options are selected). 

If you have a single ROI selected, there's no error produced in the log. If you use "entire image", you get this error:

```
INFO: Image processed in 0.00 seconds
INFO: Processing complete in 0.02 seconds
INFO: Completed!
ERROR: QuPath exception
    at qupath.lib.gui.viewer.overlays.HierarchyOverlay.lambda$paintOverlay$0(HierarchyOverlay.java:196)
    at java.base/java.util.ArrayList.removeIf(Unknown Source)
    at java.base/java.util.ArrayList.removeIf(Unknown Source)
    at qupath.lib.gui.viewer.overlays.HierarchyOverlay.paintOverlay(HierarchyOverlay.java:196)
    at qupath.lib.gui.viewer.QuPathViewer.paintViewer(QuPathViewer.java:1757)
    at qupath.lib.gui.viewer.QuPathViewer.paintCanvas(QuPathViewer.java:420)
    at qupath.lib.gui.viewer.QuPathViewerPlus.paintCanvas(QuPathViewerPlus.java:257)
    at qupath.lib.gui.viewer.QuPathViewer.lambda$repaint$4(QuPathViewer.java:508)
    at com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:428)
    at java.base/java.security.AccessController.doPrivileged(Unknown Source)
    at com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:427)
    at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
    at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
    at com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(WinApplication.java:174)
    at java.base/java.lang.Thread.run(Unknown Source)
ERROR: QuPath exception
    at qupath.lib.gui.viewer.overlays.HierarchyOverlay.lambda$paintOverlay$0(HierarchyOverlay.java:196)
    at java.base/java.util.ArrayList.removeIf(Unknown Source)
    at java.base/java.util.ArrayList.removeIf(Unknown Source)
    at qupath.lib.gui.viewer.overlays.HierarchyOverlay.paintOverlay(HierarchyOverlay.java:196)
    at qupath.lib.gui.viewer.QuPathViewer.paintViewer(QuPathViewer.java:1757)
    at qupath.lib.gui.viewer.QuPathViewer.paintCanvas(QuPathViewer.java:420)
    at qupath.lib.gui.viewer.QuPathViewerPlus.paintCanvas(QuPathViewerPlus.java:257)
    at qupath.lib.gui.viewer.QuPathViewer.lambda$repaint$4(QuPathViewer.java:508)
    at com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:428)
    at java.base/java.security.AccessController.doPrivileged(Unknown Source)
    at com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:427)
    at com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
    at com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
    at com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(WinApplication.java:174)
    at java.base/java.lang.Thread.run(Unknown Source)
INFO: 
qupath.lib.plugins.objects.TileClassificationsToAnnotationsPlugin  {"pathClass": "All classes",  "deleteTiles": false,  "clearAnnotations": false,  "splitAnnotations": false}

```

**Comment by Svidro** [comment]
And if you delete all of the annotations, select all of the detections, and run it, it plugs along for a while and ends up doing nothing. But successfully. In a lightly modified m4.
![image](https://user-images.githubusercontent.com/23145209/64463124-bc677480-d0b7-11e9-91b6-bee324dc50df.png)

**Comment by petebankhead** [comment]
Thanks, I think the commit above should fix both issues.

**Comment by petebankhead** [comment]
This should be resolved in [v0.2.0-m5](https://github.com/qupath/qupath/releases/tag/v0.2.0-m5) - please reopen if you find it isn't.
